src/code/z_actor.c

```diff
@@ -1119,15 +1119,20 @@ s32 func_8002DDF4(PlayState* play) {
     return player->stateFlags2 & PLAYER_STATE2_12;
 }
 
-void func_8002DE04(PlayState* play, Actor* actorA, Actor* actorB) {
+void Actor_SwapHookshotAttachment(PlayState* play, Actor* srcActor, Actor* destActor) {
     ArmsHook* hookshot = (ArmsHook*)Actor_Find(&play->actorCtx, ACTOR_ARMS_HOOK, ACTORCAT_ITEMACTION);
 
-    hookshot->grabbed = actorB;
-    hookshot->grabbedDistDiff.x = 0.0f;
-    hookshot->grabbedDistDiff.y = 0.0f;
-    hookshot->grabbedDistDiff.z = 0.0f;
-    actorB->flags |= ACTOR_FLAG_13;
-    actorA->flags &= ~ACTOR_FLAG_13;
+    // Note: There is no safety check for a NULL hookshot pointer.
+    // The responsibility is on the caller to make sure the hookshot exists.
```

Comment:
I think it would be better to put this comment above the function, so you get this note when you hover the function.

Reply:
The function is used once and is pretty irrelevant so I didnt bother with a function comment, but I guess someone may find a use for it someday :D